From: Sophie Brun <sophie@offensive-security.com>
Date: Tue, 6 Oct 2020 09:55:14 +0200
Subject: Restrict maximal version of thor to 0.20.3

Last-Update: 2020-10-07
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=955405
Bug: https://github.com/erikhuda/thor/issues/721

msfdb init fails with
"... thor/command.rb:2:in `<class:Thor>': superclass mismatch for class
Command (TypeError)"
We have different in incompatible versions of ruby-thor: version 1.0.1
in metasploit bundle and the packaged version 0.20.3 required by the package
bundler.
This patch can be removed when the ruby-thor package version will
be >= 1.0.0
---
 metasploit-framework.gemspec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/metasploit-framework.gemspec b/metasploit-framework.gemspec
index 8583d01..fb07063 100644
--- a/metasploit-framework.gemspec
+++ b/metasploit-framework.gemspec
@@ -91,6 +91,8 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'pg'
   # Run initializers for metasploit-concern, metasploit-credential, metasploit_data_models Rails::Engines
   spec.add_runtime_dependency 'railties'
+  # Restrict maximal version of thor
+  spec.add_runtime_dependency 'thor', '~> 0.20.3'
   # required for OS fingerprinting
   spec.add_runtime_dependency 'recog'
   # required for bitlocker fvek extraction
 
